### DISCUSSION TITLE
Annotation in UserInterface

### DISCUSSION THREAD

**Discussion by monlouisj** [symfony/symfony#44084] [Q&A] [answered]
Hi,
I expect an Interface to have some methods inside of it that you must define in your own class, but when I implement `Symfony\Component\Security\Core\User\UserInterface` , my editor points at the method `getUserIdentifier()`

This method is not inside the interface, but is written as an annotation:
` @method string getUserIdentifier() returns the identifier for this user (e.g. its username or e-mailaddress)`

I don't understand why they chose to write it this way. 
Can someone explain ?

**Comment by stof** [comment, marked as answer]
This is done this way because adding a method in an interface is a BC break, and so can only be done in Symfony 6.0 (there, it is an actual method).
As part of the migration path, we define the future methods of the interface with `@method` in phpdoc. And the DebugClassLoader will trigger deprecation warnings if you implement the interface without implementing those methods.
